table-api/src/main/java/io/deephaven/api/ColumnMatch.java

```diff
@@ -0,0 +1,33 @@
+package io.deephaven.api;
+
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Parameter;
+
+@Immutable(builder = false, copy = false)
+public abstract class ColumnMatch implements JoinMatch {
+
+    public static ColumnMatch of(ColumnName left, ColumnName right) {
+        return ImmutableColumnMatch.of(left, right);
+    }
+
+    public static ColumnMatch parse(String x) {
+        final int ix = x.indexOf('=');
+        if (ix < 0 || ix + 1 == x.length()) {
+            throw new IllegalArgumentException(String.format("Unable to parse '%s'", x));
+        }
+        final int ix2 = x.charAt(ix + 1) == '=' ? ix + 1 : ix;
```

Comment:
You're going to fail differently in the case with `=` at the end from `==` at the end.